src/commonMain/kotlin/fr/acinq/eclair/channel/Channel.kt

```diff
@@ -164,7 +167,41 @@ sealed class ChannelState {
                 this is WaitForInit && newState is Closing -> actions
                 this is Closing && newState is Closing -> actions
                 this is ChannelStateWithCommitments && newState is Closing -> {
-                    actions + ChannelAction.Storage.StoreOutgoingAmount(this.commitments.localCommit.spec.toLocal)
+                    val channelBalance = commitments.localCommit.spec.toLocal
+                    if (channelBalance > 0.msat) {
+                        val defaultScriptPubKey = commitments.localParams.defaultFinalScriptPubKey
+                        val localShutdown = when (this) {
+                            is Normal -> this.localShutdown
+                            is Negotiating -> this.localShutdown
+                            is ShuttingDown -> this.localShutdown
+                            else -> null
+                        }
+                        if (localShutdown != null && localShutdown.scriptPubKey != defaultScriptPubKey) {
+                            // Non-default output address
+                            val btcAddr = Helpers.Closing.btcAddressFromScriptPubKey(
+                                scriptPubKey = localShutdown.scriptPubKey,
+                                chainHash = staticParams.nodeParams.chainHash
+                            ) ?: "unknown"
+                            actions + ChannelAction.Storage.StoreChannelClosing(
+                                amount = channelBalance,
+                                closingAddress = btcAddr,
+                                isSentToMyWallet = false
```

Comment:
`isSentToMyWallet` indicates that the funds are sent to the default Phoenix address derived from the wallet seed. It lets the UI provide additional info and explain that the funds are still theirs, and how they can recover them. 

Typically this would be useful when the closing is a force close and the user may be confused, and would need help.

This field is also useful if the closing is mutual and Phoenix did not provide an address.

How about renaming it to `isSentToMyAddress` ?

Reply:
Gotcha, what about `isSentToDefaultAddress` then?

Reply:
Done in 443e204